### ISSUE TITLE
Rule Request: Newlines After Switch Case Bodies

### ISSUE THREAD

**Issue by mindgrub-rkoch** [nicklockwood/SwiftFormat#957]
I looked through all of the rules and implementation and could not find anything that did what I am looking for, so forgive me if there is a rule in place that will do this.

I am looking for a rule that adds a newline character after a switch case. Please see the following example:
```swift
switch self {
case .foo:
    return "foo"
case .bar:
    return "bar"
default:
    return "foobar"
}
```
The above example is somewhat hard to read, and it's a very simple example. More complex cases can make this even harder to read. Additionally, if there are some newlines between some cases, but not all, it increases the illegibility.

The proposed rule would add a newline after each case implementation if one does not already exist. The output of the above example with the rule applied would look like the following:
```swift
switch self {
case .foo:
    return "foo"

case .bar:
    return "bar"

default:
    return "foobar"
}
```

Adding the newlines can improve legibility as well as fix this instances where the there are some cases that have new lines, and others that don't.

**Comment by nicklockwood** [comment]
@mindgrub-rkoch you're right that no such rule exists currently. It shouldn't be hard to add.

**Comment by robert-koch** [comment]
@nicklockwood I'm attempting to pick this up, but had a few questions about common practices amongst the maintainers of the rules for the library.

When attempting to come up with test cases for the functionality, how far from "normal" coding practices do you typically stray and attempt to account for? As an example, here is the simple test case for this from above:
```swift
switch self {
case .foo:
    return "foo"
case .bar:
    return "bar"
default:
    return "foobar"
}
```
And here is another standard test case where a line break already exist in the switch:
```swift
switch self {
case .foo:
    return "foo"
case .bar:
    return "bar"

default:
    return "foobar"
}
```
Outside of those two test cases (or maybe a few more simple ones), would you also attempt to account for something like the following?
```swift
switch self {


case .foo:
    return "foo"
case .bar:
    return "bar" default:
        return "foobar"
}
```
*On a slight tangent:*
I've also looked through most of the implementation, and couldn't find a simple way to get the range of an entire case (including body) from within a switch. Is there a simple way, or does it involve doing a bunch of traversal of tokens to figure out what it is?

I have a preliminary implementation done, I'm just working through some optimizations and bug fixes. Any help is much appreciated.

**Comment by calda** [comment]
this is supported now
